BranchMetrics/ios-branch-deep-linking-attribution
# SDK-1796 always generate a random guid

## Reference
SDK-1796

## Summary
Add random guid that generates on app install or when tracking is reset.

Also some refactoring of existing code.
1. More unit test coverage
2. Move some utility methods from BNCDeviceInfo to BNCSystemObserver to separate class responsibilities.
3. Remove get from utility method names in BNCSystemObserver.
4. Remove unused methods.

## Motivation
This is for compatibility with some SAN apis

## Type Of Change


## Testing Instructions
Event payloads should now contain an `anon_id` guid
The value should clear when tracking is disabled
The value should change on reinstall

See ticket for sample payloads.









cc @BranchMetrics/saas-sdk-devs for visibility.
